app/src/components/pages/BookDetail/index.tsx

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Container, ThemeProvider, Typography, Grid, Link} from '@mui/material';
+import React,{useState, useEffect} from 'react';
+import Footer from '../../molecules/Footer';
+import theme from '../../Theme/theme';
+import NavBar from '../../molecules/Navbar';
+import { useLocation} from 'react-router-dom';
+import AccessTimeIcon from '@mui/icons-material/AccessTime';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import ButtonComponent from '../../atoms/Button';
+import Box from '@mui/material/Box';
+import Tab from '@mui/material/Tab';
+import TabContext from '@mui/lab/TabContext';
+import TabList from '@mui/lab/TabList';
+import TabPanel from '@mui/lab/TabPanel';
+interface BookDetails{
+  id:number;
+  title:string;
+  image:string;
+  author:string;
+  state:string;
+  state2:string;
+}
+
+const BookDetailPage=()=> {
+  const location = useLocation();
+  const index:any = location.state;
+  const [value, setValue] = React.useState('1');
+
+  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
+    setValue(newValue);
+  };
+  const [book,setBook]=useState({});
+  
+    useEffect(()=>{
+        fetch('http://localhost:8000/books/'+(index.id))
+        .then(res =>res.json())
+        .then((data:BookDetails) => {
+                setBook(data)
+            }
+            )
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    },[])
+  var thisBook:any = book;
+  var bookState = thisBook.state;
+  const handleReadNow = ()=>{
+    bookState = "Finished"
+
+    fetch('http://localhost:8000/books/'+(index.id),{
```

Comment:
You can take off all the fetch calls to a separate directory and call them wherever required.
Instead of fetch call here, call that method here. It would be more organized.